### PR TITLE
completed challenge 9 - CORE: DELETE /api/comments/:comment_id

### PR DESCRIPTION
**Pull request by Dominic20227**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -21,3 +28,23 @@ describe("CORE: GET /api/topics", () => {
...
+
+  it("404: responds with 404 message when invalid id provided", () => {
+    return request(app)
+      .delete("/api/comments/99")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toEqual("404 not found");
+      });
+  });
+});
```
Comment (popatre) [inline __tests__/app.test.js:294]: missing 400 test for bad id e.g. .delete("/api/comments/notAnId")

File: controller/controller.js [hunk 2]
```diff
@@ -5,3 +5,16 @@ exports.getAllTopics = (req, res) => {
...
   });
 };
+
+exports.deleteCommentById = (req, res, next) => {
+  const articleId = req.params.comment_id;
+
+  model
+    .deleteComment(articleId)
+    .then(() => {
+      res.status(204).send({});
```
Comment (popatre) [inline controller/controller.js]: might be cleaner to do `res.sendStatus(204)` - then dont need the empty send bit

**Review by popatre — CHANGES_REQUESTED** [review]
Missing a test:

- [ ] missing 400 test for bad id e.g. .delete("/api/comments/notAnId")

File: model/model.js [hunk 3]
```diff
@@ -5,3 +5,19 @@ exports.selectAllTopics = () => {
...
+      `DELETE FROM comments 
+                  WHERE comment_id = $1
+                  RETURNING *`,
+      [articleId]
+    )
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject("404 not found");
+      }
+      return {};
```
Comment (ghost) [inline model/model.js:115]: There's nothing to return here. You can remove this.

**Review by ghost — APPROVED** [review]
Nicely done! Go ahead and merge.

**Pull request merged**